tools/niminst/niminst.nim

```diff
@@ -672,6 +680,94 @@ proc setupDist2(c: var ConfigData) =
     else:
       quit("External program failed")
 
+type
+  TarKind {.pure.} = enum
+    ## The type of the `tar` tool
+    Invalid    ## Not a tool that is supported
+    GNU        ## GNU tar
+    Libarchive ## libarchive-based tar utility
+
+  ExternalProgramError = object of CatchableError
+    ## The error used when an external program returned a non-zero exit code
+    exitCode: int
+
+func newExternalProgramError(exitCode: int): ref ExternalProgramError =
+  result = newException(ExternalProgramError):
+    "External program exited with exit code: " & $exitCode
+  result.exitCode = exitCode
+
+proc detectTar(): (string, TarKind) =
+  ## Detect the tar utility and its type
+  const Candidates = [
+    "gtar",  # GNU tar on BSD-like platform
+    "tar",   # The regular name for the tar utility, type depends on platform
+    "bsdtar" # The libarchive tar utility
+  ]
+
+  for candidate in Candidates.items:
+    try:
+      let output = execProcess(candidate, args = ["--version"], options = {poUsePath})
+      if output.len > 0:
+        if "GNU tar" in output:
+          return (candidate, GNU)
+
+        if "libarchive" in output:
+          return (candidate, Libarchive)
+    except OSError:
+      discard "Ignore errors when the tool is not available"
+
+proc verboseExec(args: varargs[string]): int =
+  ## Print and execute command formed by `args`
+  if args.len == 0:
+    raise newException(ValueError):
+      "No command was provided for execution"
+
+  const ProcessOpts = {poUsePath, poParentStreams, poEchoCmd}
+
+  let p =
+    if args.len > 1:
```

Comment:
Are there really any savings to be had, can we not simply pass an empty array if there are no args instead?

Reply:
I'm not sure if `toOpenArray` works with empty list, hence the `if`.